Automattic/jetpack
# Instant Search: if user prefers reduced motion, always use submit overlay trigger

Fixes https://github.com/Automattic/jetpack/issues/25348.

#### Changes proposed in this Pull Request:
If user prefers reduced motion, always use the submit overlay trigger even if site preferences are for the immediate trigger.

#### Other information:


#### Does this pull request change what data or activity we track or use?
No.

#### Testing instructions:
Set up a test site with Instant Search and set the overlay trigger to 'immediate' at /wp-admin/admin.php?page=jetpack-search-configure:

<img width="269" alt="Screen Shot 2022-08-10 at 16 32 57" src="https://user-images.githubusercontent.com/17325/183816642-999d36b1-f52a-406d-bc05-3f16749098db.png">

Confirm that, on the front end of the site, typing in a search input causes the overlay to immediately open.

Now in your system accessibility settings, choose 'Reduce motion'.

<img width="780" alt="Screen Shot 2022-08-10 at 16 27 33" src="https://user-images.githubusercontent.com/17325/183816699-3407fb01-4d53-4fc8-a183-f1a62dbaeed1.png">

Confirm that, after refreshing the page on the front end of the site, typing in a search input *does not* cause the overlay to immediately open and it requires a press of "Enter" or submit button to appear (like the 'submit' overlay trigger).

#### To do:

- [x] Check media query once per component mount
- [ ] ~Add listener to check for changes to value~
